test/unittests/exportable_fixture.hpp

```diff
@@ -0,0 +1,18 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2024 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include <gtest/gtest.h>
+
+namespace evmone::test
+{
+class ExportableFixture : public testing::Test
+{
+protected:
+    std::unique_ptr<std::ostream> export_out;
```

Comment:
This doesn't work, it has to be `std::unique_ptr<std::ofstream>` otherwise proper file stream destructor is not called and file is not written to disk.

Reply:
What a hell? Am I doing something wrong?

Reply:
Still not sure why it doesn't work. But this is fixed differently now because we also want to avoid creating empty files. So the file creation is left up to the concrete test suite.

I also added a cmake test that verifies an example file was created during export.

Reply:
> Still not sure why it doesn't work

I'm also not sure anymore, becaue apparently destructor is indeed virtual :man_shrugging:

Reply:
I was able to reproduce the same issue.